charts/consul/templates/connect-inject-deployment.yaml

```diff
@@ -11,6 +11,7 @@
 {{- $dnsRedirectionEnabled := (or (and (ne (.Values.dns.enableRedirection | toString) "-") .Values.dns.enableRedirection) (and (eq (.Values.dns.enableRedirection | toString) "-") .Values.connectInject.transparentProxy.defaultEnabled)) -}}
 {{ template "consul.validateRequiredCloudSecretsExist" . }}
 {{ template "consul.validateCloudSecretKeys" . }}
+{{ template "consul.validateResourceAPIs" . }}
```

Comment:
IMO the tests for this can move to `helpers.bats` and hence you only need the assertions for `-resource-api=true` in `connect-inject-deployment.bats`

Reply:
Oh, great suggestion! I didn't even notice that `helpers` had it's own set of tests.